File: commons/dto/src/main/java/net/blueshell/dto/TelemetryDTO.java
package net.blueshell.dto;


import lombok.Data;
import lombok.EqualsAndHashCode;
import net.blueshell.enums.PlatformType;

import java.sql.Timestamp;

@Data
@EqualsAndHashCode(callSuper = true)
public class TelemetryDTO extends BaseDTO {
    private String id;
    private String url;
    private PlatformType platform;
    private Timestamp createdAt;
}

File: api/src/main/kotlin/net/blueshell/api/domain/telemetry/web/dto/TelemetryDTO.kt
package net.blueshell.api.domain.telemetry.web.dto

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import net.blueshell.api.shared.dto.AuditedAutoIdDTO
import net.blueshell.api.shared.enums.PlatformType

@Schema(name = "Telemetry")
data class TelemetryDTO(
    @field:NotBlank
    var url: String? = null,

    @field:NotNull
    var platform: PlatformType? = null
) : AuditedAutoIdDTO()
